manage/sawtooth_manage/vnm.py

```diff
@@ -23,8 +25,20 @@ def __init__(self,
 
     def update(self):
         commands = self._node_command_generator.get_commands()
+
+        # execute commands in parallel
+        processes = []
         for command in commands:
-            command.execute(self._node_controller)
+            process = Process(
+                target=command.execute,
+                args=(self._node_controller,))
+            process.start()
+            processes.append(process)
+
+        # block until all processes have completed or timed out
+        wait_time = 30
```

Comment:
If you wanted to bound the time by a total of 30s, not 30s*len(processes), then you would want a different approach.  One similar problem is solved here:

https://github.com/vaporos/sawtooth-core/commit/ddae6c479b2cab210f3ae1f699358537608c785f#diff-935cde9b4842b2d10086c6304d39945bR315

That's not necessarily a perfect solution either, but you can see how it would be easy to bound the loop to a maximum number of iterations.